File: data/derived_cot/rq3_prompting/results/translation/r1/java_to_cpp/code_output/SignInSystem.cpp
#include <unordered_map>
#include <string>
#include <vector>

class SignInSystem {
private:
    std::unordered_map<std::string, bool> users;

public:
    SignInSystem() = default;

    bool addUser(const std::string& username) {
        if (users.find(username) != users.end()) {
            return false;
        }
        users[username] = false;
        return true;
    }

    bool signIn(const std::string& username) {
        auto it = users.find(username);
        if (it == users.end()) {
            return false;
        }
        it->second = true;
        return true;
    }

    bool checkSignIn(const std::string& username) {
        auto it = users.find(username);
        if (it == users.end()) {
            return false;
        }
        return it->second;
    }

    bool allSignedIn() {
        for (const auto& entry : users) {
            if (!entry.second) {
                return false;
            }
        }
        return true;
    }

    std::vector<std::string> allNotSignedIn() {
        std::vector<std::string> result;
        for (const auto& entry : users) {
            if (!entry.second) {
                result.push_back(entry.first);
            }
        }
        return result;
    }
};

File: data/derived_cot/rq1_traces/translation/glm5.1/cpp_to_py/code_output/SignInSystem.py
class SignInSystem:
    def __init__(self):
        self.users = {}

    def add_user(self, username):
        if username in self.users:
            return False
        else:
            self.users[username] = False
            return True

    def sign_in(self, username):
        if username not in self.users:
            return False
        else:
            self.users[username] = True
            return True

    def check_sign_in(self, username):
        if username not in self.users:
            return False
        else:
            return self.users[username]

    def all_signed_in(self):
        for signed_in in self.users.values():
            if not signed_in:
                return False
        return True

    def all_not_signed_in(self):
        return [username for username, signed_in in self.users.items() if not signed_in]